architecture/core.rst

```diff
@@ -651,6 +651,110 @@ is mapped new data is copied from backing store and the hash of the page is
 verified. If it is OK the pager returns from the exception to resume the
 execution.
 
+Data structures
+===============
+.. figure:: ../images/core/tee_pager_area.png
```

Comment:
I gave it a try using Graphviz, since that has the benefit of having the actual graph directly in the documentation instead of relying on other external tools and the need for storing the source (*.dia) nor the generated png. Having that said, I think your image is looking better. So the choice is whether to have more pleasant image to look at, at the cost of relying on external tools. What's your thought here?

```suggestion
Diagram showing how the main pager data structures relates to each other:                                                                                                                                          
                                                                                                                                                                                                                   
.. graphviz::                                                                                                                                                                                                      
                                                                                                                                                                                                                   
    digraph tee_pager_area {                                                                                                                                                                                       
        graph [                                                                                                                                                                                                    
            rankdir = "UD"                                                                                                                                                                                         
        ];                                                                                                                                                                                                         
                                                                                                                                                                                                                   
        "tee_pager_area_head" [                                                                                                                                                                                    
            label = "<f0> tee_pager_area_head"                                                                                                                                                                     
            shape = "record"                                                                                                                                                                                       
        ];                                                                                                                                                                                                         
                                                                                                                                                                                                                   
        "tee_pager_area_1" [                                                                                                                                                                                       
            label = "<f0> tee_pager_area"                                                                                                                                                                          
            shape = "record"                                                                                                                                                                                       
        ];                                                                                                                                                                                                         
                                                                                                                                                                                                                   
        "tee_pager_area_2" [                                                                                                                                                                                       
            label = "<f0> tee_pager_area"                                                                                                                                                                          
            shape = "record"                                                                                                                                                                                       
        ];                                                                                                                                                                                                         
                                                                                                                                                                                                                   
        "tee_pager_area_3" [                                                                                                                                                                                       
            label = "<f0> tee_pager_area"                                                                                                                                                                          
            shape = "record"                                                                                                                                                                                       
        ];                                                                                                                                                                                                         
                                                                                                                                                                                                                   
        "fobj_1" [                                                                                                                                                                                                 
            label = "<f0> fobj"                                                                                                                                                                                    
            shape = "record"                                                                                                                                                                                       
        ];                                                                                                                                                                                                         
                                                                                                                                                                                                                   
        "fobj_2" [                                                                                                                                                                                                 
            label = "<f0> fobj"                                                                                                                                                                                    
            shape = "record"                                                                                                                                                                                       
        ];                                                                                                                                                                                                         
                                                                                                                                                                                                                   
        "tee_pager_mem_1" [                                                                                                                                                                                        
            label = "<f0> tee_pager_mem"                                                                                                                                                                           
            shape = "record"                                                                                                                                                                                       
        ];                                                                                                                                                                                                         
                                                                                                                                                                                                                   
        "tee_pager_mem_2" [                                                                                                                                                                                        
            label = "<f0> tee_pager_mem"                                                                                                                                                                           
            shape = "record"                                                                                                                                                                                       
        ];                                                                                                                                                                                                         
                                                                                                                                                                                                                   
        "tee_pager_mem_3" [                                                                                                                                                                                        
            label = "<f0> tee_pager_mem"                                                                                                                                                                           
            shape = "record"                                                                                                                                                                                       
        ];
        "tee_pager_area_head":f0 -> "tee_pager_area_1":f0                                                                                                                                                          
                                                                                                                                                                                                                   
        "tee_pager_area_1":f0 -> "fobj_1":f0                                                                                                                                                                       
        "tee_pager_area_1":f0 -> "tee_pager_area_2":f0                                                                                                                                                             
        "tee_pager_area_2":f0 -> "fobj_1":f0                                                                                                                                                                       
        "tee_pager_area_2":f0 -> "tee_pager_area_3":f0                                                                                                                                                             
        "tee_pager_area_3":f0 -> "fobj_2":f0                                                                                                                                                                       
                                                                                                                                                                                                                   
        "tee_pager_mem_1":f0 -> "fobj_1":f0                                                                                                                                                                        
        "tee_pager_mem_1":f0 -> "tee_pager_mem_2":f0                                                                                                                                                               
        "tee_pager_mem_2":f0 -> "tee_pager_mem_3":f0                                                                                                                                                               
        "tee_pager_mem_3":f0 -> "fobj_2":f0                                                                                                                                                                        
    }
```

Reply:
In case you don't want to re-render the docs, but still see the result, ... here it is.
![image](https://user-images.githubusercontent.com/6168937/53487951-3bd31600-3a8d-11e9-8752-ea32eeb57531.png)

Reply:
I don't think this is good enough. I spent almost a day trying to get graphviz to generate something remotely like what I ended up doing in dia.